ui/src/components/TermDisplay.vue

```diff
@@ -8,30 +8,68 @@
 import { Vue, Component, Prop } from 'vue-property-decorator'
 import { Term } from 'rdf-js'
 import { shrink } from '@/rdf-properties'
+import { rdfs, schema } from '@tpluscode/rdf-ns-builders'
+import { GraphPointer } from 'clownface'
 
 @Component
 export default class extends Vue {
-  @Prop({ required: true }) term!: Term
+  @Prop({ required: true }) term!: Term | GraphPointer
   @Prop({ default: false }) showLanguage!: boolean
   @Prop() base?: string
 
-  get displayShort (): string {
-    if (this.term.termType === 'NamedNode') {
-      return shrink(this.term.value, this.base)
-    } else {
-      return this.term.value
+  get displayShort () {
+    return this.__schemaName || this.__commonTermPrefixes || this.__rawLabel
+  }
+
+  mounted (): void {
+    const label = this.__schemaName || this.__commonTermPrefixes
+    if (!label) {
+      this.$emit('no-label')
     }
   }
 
+  get node (): Term {
+    if ('_context' in this.term) {
+      return this.term.term
+    }
+    return this.term
+  }
+
   get displayFull (): string {
-    if (this.term.termType === 'Literal') {
-      const datatype = this.term.datatype ? `^^${shrink(this.term.datatype.value)}` : ''
-      const language = this.term.language ? `@${this.term.language}` : ''
+    if (this.node.termType === 'Literal') {
+      const datatype = this.node.datatype ? `^^${shrink(this.node.datatype.value)}` : ''
+      const language = this.node.language ? `@${this.node.language}` : ''
 
-      return `"${this.term.value}${language}"${datatype}`
+      return `"${this.node.value}${language}"${datatype}`
     } else {
-      return this.term.value
+      return this.node.value
+    }
+  }
+
+  get __commonTermPrefixes (): string | null {
+    const shrunk = shrink(this.node.value)
+
+    return shrunk !== this.node.value ? shrunk : null
+  }
+
+  get __schemaName (): string | null {
+    const pointer = '_context' in this.term
+      ? this.term
+      : null
+
+    if (pointer) {
+      return pointer.out([schema.name, rdfs.label], { language: ['en', '*'] }).values[0]
```

Comment:
I still have the same issue: `TermDisplay` was never supposed to display a label for a resource. It's supposed to display a term. I think it would be better to not touch it and implement what you need in `ExternalTerm`.